components/bl-mapbox-component/src/map.js

```diff
@@ -0,0 +1,232 @@
+import { useEffect, useRef, useState } from 'react';
+
+import mapboxgl from './lib/mapbox';
+import MapboxDirections from './lib/mapbox-directions';
+import MapboxGeocoder from './lib/mapbox-geocoder';
+import { createActions } from './actions';
+import { preparePolygons, applyFog } from './helpers';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function Map({ component, eventHandlers, settings }) {
+  const mapContainerRef = useRef(null);
+  const mapRef = useRef(null);
+
+  const [polygonsArray, setPolygonsArray] = useState([]);
+  const [markersArray, setMarkersArray] = useState([]);
+
+  const { accessToken } = settings;
+
+  const { onMarkerClick, onPolygonClick, onDeterminingGeoposition } = eventHandlers;
+
+  const { markers, polygons, center, fullScreen, navigation, searchBar, geolocation, classList } = component;
+
+  useEffect(() => {
+    mapboxgl.accessToken = accessToken;
+
+    initMapboxLibrary(mapRef, mapContainerRef, component);
+  }, []);
+
+  useEffect(() => {
+    if (mapRef.current && center) {
+
+      mapRef.current.setCenter([center.lng, center.lat]);
+    }
+  }, [center]);
+
+  useEffect(() => {
+    mapRef.current.on('load', () => {
+
+      applyFog(mapRef, component);
+
+      useEvents(mapRef, eventHandlers);
+
+      if (searchBar) {
+        mapRef.current.addControl(
+          new MapboxGeocoder({
+            accessToken: mapboxgl.accessToken,
+            mapboxgl   : mapboxgl,
+          })
+        );
+      }
+
+      if (fullScreen) {
+        mapRef.current.addControl(new mapboxgl.FullscreenControl());
+      }
+
+      if (navigation) {
+        mapRef.current.addControl(new mapboxgl.NavigationControl());
+      }
+
+      if (geolocation) {
+        useGeolocation(mapRef, onDeterminingGeoposition);
+      }
+
+      createActions(mapRef, component);
+    });
+  }, []);
+
+  useEffect(() => {
+    useMarkers(markers, markersArray, setMarkersArray, mapRef, onMarkerClick);
+  }, [markers]);
+
+  useEffect(() => {
+    usePolygons(polygons, polygonsArray, setPolygonsArray, mapRef, onPolygonClick);
+  }, [polygons]);
+
+  return (
+    <div>
```

Comment:
this empty div needed?

Reply:
Yes, it doesn't work without it.

Reply:
@addKiwi could you please clarify what exactly does not work?

Reply:
when I remove `div`,  I  don't see difference in HTML tags, but map is not displaying and `div` have 0 height

Reply:
If I not using refs it works without `div`
[Example](https://docs.mapbox.com/help/tutorials/use-mapbox-gl-js-with-react/) with ref

Reply:
ok